src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -643,25 +638,14 @@ private void DisconnectChainHeader(ChainedHeader header)
         }
 
         /// <inheritdoc />
-        public ChainedHeader FindHeaderAndVerifyBlockIntegrity(Block block)
+        public ChainedHeader FindHeader(uint256 blockHash)
         {
-            uint256 blockHash = block.GetHash();
-            this.logger.LogTrace("({0}:'{1}')", nameof(block), blockHash);
-
-            ChainedHeader chainedHeader;
+            this.logger.LogTrace("({0}:'{1}')", nameof(blockHash), blockHash);
 
-            if (!this.chainedHeadersByHash.TryGetValue(blockHash, out chainedHeader))
+            if (!this.chainedHeadersByHash.TryGetValue(blockHash, out ChainedHeader chainedHeader))
```

Comment:
why is this not throwing an exception anymore?

Reply:
>why is this not throwing an exception anymore?

Before it made sense to throw because we would also validate the header. 

Now the method just looks up the header so I thought it's a good idea to return null because we have similar behavior in other methods in the codebase that do the lookups and have same naming pattern